From 2bcc788efba34494ceb5e57df65a891a3df3aea3 Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf@emagii.com>
Date: Sun, 19 Feb 2023 10:51:19 +0100
Subject: [PATCH 03/13] ChangeLog for ASCII, ASCIZ linker commands

Signed-off-by: Ulf Samuelsson <ulf@emagii.com>
---
 ld/ChangeLog | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/ld/ChangeLog b/ld/ChangeLog
index 3be5c5c7..15980fd5 100644
--- a/ld/ChangeLog
+++ b/ld/ChangeLog
@@ -1,3 +1,25 @@
+2023-02-16  Ulf Samuelsson <binutils@emagii.com>
+	    Nick Clifton  <nickc@redhat.com>
+
+	* ldlex.l: Add ASCII token.
+	* ldgram.y: Add parsing of the ASCII command.
+	* ldlang.c (lang_add_string): Add maximum size parameter.  Move
+	escape character handling code into separate function.
+	* ldlang.h (lang_add_string): Update prototype.
+	* NEWS: Mention the new feature.
+	* ld.texi (Output Section Data): Document the new directives.
+	* testsuite/ld-scripts/asciz.t: Adjust to work on more architectures
+	and to test more aspects of the ASCIZ directive.
+	* testsuite/ld-scripts/asciz.d: Adjust to match the changes to the
+	test linker script.
+	* testsuite/ld-scripts/ascii.d: New test driver.
+	* testsuite/ld-scripts/ascii.s: New test assembler source.
+	* testsuite/ld-scripts/ascii.t: New test script.
+	* testsuite/ld-scripts/script.exp: Run the new test.
+
+	* The linker script syntax has a new command for output sections: ASCIZ "string"
+	This will insert a zero-terminated string at the current location.
+
 2022-02-17  Nick Clifton  <nickc@redhat.com>
 
 	* po/sr.po: Updated Serbian translation.
-- 
2.17.1

